crates/oxc_linter/src/service.rs

```diff
@@ -121,37 +122,63 @@ impl Runtime {
             resolver: Self::resolver(),
             module_map: ModuleMap::default(),
             cache_state: CacheState::default(),
+            partial_loaders: FxHashMap::default(),
```

Comment:
How about having `partial_vue_loader` and `partial_astro_loader` expanded out here, and then do manual checks in the runner?

```
if ext == "vue" {
source_text = partial_vue_loader.parse()
}
```